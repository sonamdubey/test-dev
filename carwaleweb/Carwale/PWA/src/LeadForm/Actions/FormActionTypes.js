//Action type for form
export const SET_BUYER_DATA = "SET_BUYER_DATA";
export const SET_SELLER_DATA = "SET_SELLER_DATA";
export const SET_LOCATION = "SET_LOCATION";
export const SET_EMAIL = "SET_EMAIL";
export const SET_MLA_DATA = "SET_MLA_DATA";
export const SET_RECOMMENDATION_DATA = "SET_RECOMMENDATION_DATA";

//Action type for setting encryptedLeadId
export const SET_LEAD_ID = "SET_LEAD_ID";
export const CLEAR_LEAD_ID = "CLEAR_LEAD_ID";

export const PREFILL_LOCATION = "PREFILL_LOCATION";
export const SET_CAMPAIGN = "SET_CAMPAIGN";
export const SET_CAMPAIGN_MODEL_KEY = "SET_CAMPAIGN_MODEL_KEY";

export const SET_INTERACTION_ID = "SET_INTERACTION_ID";

export const SET_OTHERS = "SET_OTHERS";
