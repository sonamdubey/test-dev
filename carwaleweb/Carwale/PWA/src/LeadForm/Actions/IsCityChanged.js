export const SET_CITY_CHANGE = "SET_CITY_CHANGE";
