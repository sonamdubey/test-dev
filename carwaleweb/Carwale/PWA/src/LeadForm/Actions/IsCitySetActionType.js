export const SET_LEAD_SOURCE_CITY = "SET_LEAD_SOURCE_CITY";
