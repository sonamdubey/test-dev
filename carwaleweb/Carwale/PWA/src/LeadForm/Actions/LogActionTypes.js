//Action type for setting Logs
export const SET_INFO_LOG = "SET_INFO_LOG";
export const SET_ERROR_LOG = "SET_ERROR_LOG";
export const SET_DEBUG_LOG = "SET_DEBUG_LOG";
export const CLEAR_LOG = "CLEAR_LOG";
