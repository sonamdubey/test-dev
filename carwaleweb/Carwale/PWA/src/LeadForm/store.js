import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";

import rootReducer from "./Reducers/index";

class Store {
  constructor() {
    this.store = null;
  }

  createInstance() {
    const middleware = [thunk];
    let storeEnhancer;
    if (__DEV__) {
      const { logger } = require("redux-logger");
      middleware.push(logger);
      const { composeWithDevTools } = require("redux-devtools-extension");
      storeEnhancer = composeWithDevTools(applyMiddleware(...middleware));
    } else {
      storeEnhancer = applyMiddleware(...middleware);
    }

    this.store = createStore(rootReducer, storeEnhancer);
  }

  getInstance() {
    if (this.store == null) {
      this.createInstance();
    }
    return this.store;
  }
}

const store = new Store().getInstance();

export default store;
