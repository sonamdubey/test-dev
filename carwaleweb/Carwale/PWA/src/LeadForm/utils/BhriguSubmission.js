import { mapBhriguLogs } from "./ObjectMapping";
import bhriguLogSubmissionApi from "../Api/BhriguLogSubmission";

export function submitBhriguLogs(logData) {
  let bhriguEvents = mapBhriguLogs(logData);
  bhriguLogSubmissionApi
    .set(bhriguEvents)
    .then(status => {})
    .catch(error => {
      console.log("Bhrigu Submission Failed, " + error);
    });
}
