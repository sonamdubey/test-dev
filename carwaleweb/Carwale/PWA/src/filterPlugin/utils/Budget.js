export const formatBudgetTooltipValue = (value) => {
	let budgetValue

	if (value > 9999999) {
		budgetValue = parseFloat((value / 10000000).toFixed(2)) + '+ Cr'
	}
	else if (value <= 200000) {
		budgetValue = 'Upto 2 lakh'
	}
	else {
		budgetValue = parseFloat((value / 100000).toFixed(2)) + ' lakh'
	}

	return budgetValue
}
