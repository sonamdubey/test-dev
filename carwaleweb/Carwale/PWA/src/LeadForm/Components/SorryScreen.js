import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { submitBhriguLogs } from "../utils/BhriguSubmission";
import { clearLog } from "../ActionCreators/LogState";
import store from "../store";

class SorryScreen extends PureComponent {
  componentDidMount() {
    this.props.setLog();
    submitBhriguLogs(store.getState().log);
    store.dispatch(clearLog());
  }

  render() {
    return (
      <div className="sorry-screen-div">
        <div className="location-popup__sorry-icon-wrapper">
          <span className="location-popup-icon location-popup__sorry-icon" />
        </div>
        <p className="sorry-screen__response">
          {"We couldn't find any dealerships in your city"}
        </p>
      </div>
    );
  }
}

SorryScreen.propTypes = {
  setLog: PropTypes.func
};

export default SorryScreen;
