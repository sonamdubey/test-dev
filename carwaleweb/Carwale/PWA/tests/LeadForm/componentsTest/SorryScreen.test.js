import React from "react";
import SorryScreen from "../../../src/LeadForm/Components/SorryScreen";
import store from "../../../src/LeadForm/store";
import { submitBhriguLogs } from "../../../src/LeadForm/utils/BhriguSubmission";
import { CLEAR_LOG } from "../../../src/LeadForm/Actions/LogActionTypes";

jest.mock("../../../src/LeadForm/utils/BhriguSubmission", () => {
  return { submitBhriguLogs: jest.fn() };
});

jest.mock("../../../src/LeadForm/ActionCreators/ScreenState", () => {
  return {
    clearLog: jest.fn(() => ({
      type: "CLEAR_LOG"
    }))
  };
});

jest.mock("../../../src/LeadForm/store", () => {
  return mockStore({
    log: [{}]
  });
});

let props = { setLog: jest.fn() };
let wrapper;
beforeEach(() => {
  wrapper = shallowWithStore(<SorryScreen {...props} />, store);
});

describe("Testing SorryScreen Component - DOM", () => {
  test("Component SorryScreen Snapshot RenderedTree", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Testing SorryScreen Component - ComponentDidMount", () => {
  test("'setlog' of props called - sorry screen mounted", () => {
    expect(props.setLog).toHaveBeenCalled();
  });

  test("SubmitBhriguLogs called - sorry screen mounted", () => {
    expect(submitBhriguLogs).toHaveBeenCalledWith([{}]);
  });

  test("Logs Cleared - sorry screen mounted", () => {
    let clearLog = { type: CLEAR_LOG };
    let actions = store.getActions();
    expect(actions).toContainEqual(clearLog);
  });
});
