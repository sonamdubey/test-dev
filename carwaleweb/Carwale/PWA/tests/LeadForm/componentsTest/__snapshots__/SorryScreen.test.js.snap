// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SorryScreen Component - DOM Component SorryScreen Snapshot RenderedTree 1`] = `
<div
  className="sorry-screen-div"
>
  <div
    className="location-popup__sorry-icon-wrapper"
  >
    <span
      className="location-popup-icon location-popup__sorry-icon"
    />
  </div>
  <p
    className="sorry-screen__response"
  >
    We couldn't find any dealerships in your city
  </p>
</div>
`;
